src/main/kotlin/xyz/crossward/service/StatsService.kt

```diff
@@ -0,0 +1,33 @@
+package xyz.crossward.service
+
+import org.springframework.stereotype.Service
+import xyz.crossward.exception.BadRequestException
+import xyz.crossward.repository.EntryRepository
+import xyz.crossward.repository.UserRepository
+
+
+@Service
+class StatsService(
+    private val entryRepository: EntryRepository,
+    private val userRepository: UserRepository
+) {
+    fun getAverageTimeByUserId(userId: String): Double {
+        if (!userRepository.existsById(userId)) {
+            throw BadRequestException("A user with ID ${userId} does not exist")
+        }
+        val average = entryRepository.findTimesByUserId(userId).toList().average()
+        if (average.isNaN())
+        {
+            throw BadRequestException("No entries found for user with ID ${userId}")
+        }
+        return average
+    }
+
+    fun getBestTimeByUserId(userId: String): Int {
+        if (!userRepository.existsById(userId)) {
+            throw BadRequestException("A user with ID ${userId} does not exist")
+        }
+        val times = entryRepository.findTimesByUserId(userId)
+        return times.min(Integer::compare).orElseThrow { throw BadRequestException("No entries found for user with ID ${userId}")}
```

Comment:
I think the more kotlin way to write this would be like this:
```Kotlin
times.minOrNull() ?: throw BadRequestException("No entries found for user with ID ${userId}")}
```
and could be shortened:
```Kotlin
return entryRepository.findTimesByUserId(userId)
    .minOrNull() ?: throw BadRequestException("No entries found for user with ID ${userId}")}
```

Reply:
Kotlin doesn't have a .minOrNull for streams, only Lists I think

Reply:
Ok see if you can throw a toList() and then do .minOrNull()